misc.go

```diff
@@ -886,6 +866,23 @@ var hintTokenMap = map[string]int{
 	"MATERIALIZATION": hintMaterialization,
 }
 
+type FeaturesMap map[string]struct{}
```

Comment:
does this type need to be public?

Reply:
@kennytm I came up with an idea: how about putting the `FeaturesMap` into a part of the parser driver? Let users decide what features should be parsed.